packages/react/src/GadgetProvider.tsx

```diff
@@ -1,22 +1,98 @@
-import type { GadgetConnection } from "@gadgetinc/api-client-core";
-import { $gadgetConnection } from "@gadgetinc/api-client-core";
+import type { AnyClient, GadgetConnection } from "@gadgetinc/api-client-core";
+import { $gadgetConnection, isGadgetClient } from "@gadgetinc/api-client-core";
+import type { ReactNode } from "react";
 import React from "react";
-import type { Client } from "urql";
-import { Provider } from "urql";
+import type { Client as UrqlClient } from "urql";
+import { Provider as UrqlProvider } from "urql";
+
+/**
+ * React context that stores the current urql client
+ *
+ * urql doesn't have its own useClient hook, so we store it on our own context to get at the client object later
+ **/
+export const GadgetUrqlClientContext = React.createContext<UrqlClient | undefined>(undefined);
+
+/**
+ * React context that stores an instance of the JS Client for an app (AKA the `api` object)
+ */
+export const GadgetClientContext = React.createContext<AnyClient | undefined>(undefined);
```

Comment:
Doesn't seem like this context is consumed anywhere? This is just to match the above urql context I'm assuming.